Sofa/GUI/Qt/src/sofa/gui/qt/MaterialDataWidget.cpp

```diff
@@ -177,8 +177,8 @@ bool VectorMaterialDataWidget::createWidgets()
     layout->addWidget(_comboBox);
     layout->addWidget(_materialDataWidget);
 
-    connect( _comboBox, SIGNAL(activated(int)  ), this, SLOT( changeMaterial( int) ) );
-    connect( _materialDataWidget, SIGNAL( WidgetDirty(bool) ) ,this, SLOT( setWidgetDirty(bool) ) );
+    connect( _comboBox, QOverload<int>::of(&QComboBox::currentIndexChanged), this, &VectorMaterialDataWidget::changeMaterial );
```

Comment:
What is the logic of using `QOverload`?

Reply:
With the new connect syntax, if a method has multiple overloads (like currentIndexChanged) then the type deduction mechanisme cannot select which one to use. 

This is explained in: https://doc.qt.io/qt-5/qcombobox.html#currentIndexChanged